packages/aws-cdk/lib/api/cloudformation-deployments.ts

```diff
@@ -372,6 +371,123 @@ export class CloudFormationDeployments {
+    return (await this.prepareSdkFor(stackArtifact, undefined, Mode.ForReading)).stackSdk;
+  }
+
+  private async readCurrentStackTemplate(stackName: string, stackSdk: ISDK) : Promise<Template> {
+    // if `stackName !== stackArtifact.stackName`, then `stackArtifact` is an ancestor to a nested stack with name `stackName`.
+    const cfn = stackSdk.cloudFormation();
+    const stack = await CloudFormationStack.lookup(cfn, stackName);
+    return stack.template();
+  }
+
+  private async replaceNestedStacksInRootStack(
+    rootStackArtifact: cxapi.CloudFormationStackArtifact, deployedTemplate: any, sdk: ISDK,
+  ): Promise<void> {
+    await this.replaceNestedStacksInParentTemplate(
+      rootStackArtifact, rootStackArtifact.template, deployedTemplate, rootStackArtifact.stackName, sdk,
+    );
+  }
+
+  private async replaceNestedStacksInParentTemplate(
+    rootStackArtifact: cxapi.CloudFormationStackArtifact,
+    generatedParentTemplate: any,
+    deployedParentTemplate: any,
+    parentStackName: string | undefined,
+    sdk: ISDK,
+  ): Promise<void> {
+    const listStackResources = parentStackName ? new LazyListStackResources(sdk, parentStackName) : undefined;
+    const parentStackResources = Object.entries(generatedParentTemplate.Resources ?? {});
+    for (const [nestedStackLogicalId, nestedStackResource] of parentStackResources) {
+      if (!this.isCdkManagedNestedStack(nestedStackResource)) {
+        continue;
+      }
+
+      const assetPath = nestedStackResource.Metadata['aws:asset:path'];
+      const nestedStackTemplates = await this.getNestedStackTemplates(
+        rootStackArtifact, assetPath, nestedStackLogicalId, parentStackName, listStackResources, sdk,
+      );
+
+      if (!generatedParentTemplate.Resources[nestedStackLogicalId].Properties) {
+        generatedParentTemplate.Resources[nestedStackLogicalId].Properties = {};
+      }
+      generatedParentTemplate.Resources[nestedStackLogicalId].Properties.NestedTemplate = nestedStackTemplates.generatedNestedTemplate;
+
+      if (!deployedParentTemplate.Resources) {
+        deployedParentTemplate.Resources = {};
+      }
+      if (!deployedParentTemplate.Resources[nestedStackLogicalId]) {
+        deployedParentTemplate.Resources[nestedStackLogicalId] = {};
+        deployedParentTemplate.Resources[nestedStackLogicalId].Type = 'AWS::CloudFormation::Stack';
+      }
+      if (!deployedParentTemplate.Resources[nestedStackLogicalId].Properties) {
+        deployedParentTemplate.Resources[nestedStackLogicalId].Properties = {};
+      }
+      deployedParentTemplate.Resources[nestedStackLogicalId].Properties.NestedTemplate = nestedStackTemplates.deployedNestedTemplate;
+
+      await this.replaceNestedStacksInParentTemplate(
+        rootStackArtifact,
+        generatedParentTemplate.Resources[nestedStackLogicalId].Properties.NestedTemplate,
+        deployedParentTemplate.Resources[nestedStackLogicalId].Properties.NestedTemplate ?? {},
```

Comment:
Same here:

```suggestion
        nestedStackTemplates.deployedNestedTemplate,
```

(I'm 99% sure that `??` is dead code)